qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -20,139 +20,20 @@
- */
-template <typename Map>
-Map combineParameterValues(const Map &parameterValues1,
-                           const Map &parameterValues2) {
-  auto combinedValues = parameterValues1;
-  for (const auto &index : parameterValues2.keys())
-    combinedValues[index] = parameterValues2[index];
-  return combinedValues;
-}
-
-/**
- * Reverts the specified changes made to the specified map of values.
- *
- * @param map     The map to apply the reversion to.
- * @param changes The list of changes (map-like structure), detailing
- *                the value before and after the change.
- */
-template <typename Map, typename Changes>
-void revertChanges(Map &map, const Changes &changes) {
-  for (const auto &beforeChange : changes.keys()) {
-    const auto &afterChange = changes[beforeChange];
-
-    for (auto &values : map) {
-      if (values.contains(afterChange)) {
-        const auto value = values[afterChange];
-        values.remove(afterChange);
-        values[beforeChange] = value;
-      }
-    }
-  }
-}
-
-/**
- * @return  True if the first function precedes the second when ordering by
- *          name.
- */
-bool functionNameComparator(IFunction_const_sptr first,
-                            IFunction_const_sptr second) {
-  return first->name() < second->name();
-}
-
-/**
- * Extracts the functions from a composite function into a vector.
- *
- * @param composite The composite function.
- * @return          A vector of the functions in the specified composite
- *                  function.
- */
-std::vector<IFunction_const_sptr>
-extractFunctions(const CompositeFunction &composite) {
-  std::vector<IFunction_const_sptr> functions;
-  functions.reserve(composite.nFunctions());
-
-  for (auto i = 0u; i < composite.nFunctions(); ++i)
-    functions.emplace_back(composite.getFunction(i));
-  return functions;
-}
-
-/*
- * Checks whether the specified composite functions have the same composition.
- *
- * @param composite1 Function to compare.
- * @param composite2 Function to compare.
- * @return           True if the specified functions have the same composition,
- *                   False otherwise.
- */
-bool equivalentComposites(const CompositeFunction &composite1,
-                          const CompositeFunction &composite2) {
-
-  if (composite1.nFunctions() != composite2.nFunctions()) {
-    return false;
-  } else {
-    auto functions1 = extractFunctions(composite1);
-    auto functions2 = extractFunctions(composite2);
-    std::sort(functions1.begin(), functions1.end(), functionNameComparator);
-    std::sort(functions2.begin(), functions2.end(), functionNameComparator);
-
-    for (auto i = 0u; i < functions1.size(); ++i) {
-      if (!equivalentFunctions(functions1[i], functions2[i]))
-        return false;
+void updateParameters(
+    IFunction_sptr function,
+    const std::unordered_map<std::string, ParameterValue> &parameters) {
+  for (auto i = 0u; i < function->nParams(); ++i) {
```

Comment:
Is it possible here to use `for(auto && name : function->getParameterNames())`, and use the overloaded `setParameter` which takes the name?

Actually, not sure if there is an appropriate `setError` function for this case.

Reply:
This is a possibility -- the problem is that the setParameter method which takes a name rather than an index, goes onto find the index itself -- this would be slower since I would need to find the index of each parameter on every iteration.